Sources/Sentry/SentryFileManager.m

```diff
@@ -312,15 +312,35 @@ - (void)removeFileAtPath:(NSString *)path
 
 - (NSString *)storeEnvelope:(SentryEnvelope *)envelope
 {
-    NSData *envelopeData = [SentrySerialization dataWithEnvelope:envelope];
-
     @synchronized(self) {
         NSString *path =
             [self.envelopesPath stringByAppendingPathComponent:[self uniqueAscendingJsonName]];
+
         SENTRY_LOG_DEBUG(@"Writing envelope to path: %@", path);
 
-        if (![self writeData:envelopeData toPath:path]) {
-            SENTRY_LOG_WARN(@"Failed to store envelope.");
+        [[NSFileManager defaultManager] createFileAtPath:path contents:nil attributes:nil];
+
+        NSFileHandle *fileHandle = [NSFileHandle fileHandleForWritingAtPath:path];
+
+        if (!fileHandle) {
+            SENTRY_LOG_ERROR(@"Couldn't get NSFileHandle for path: %@", path);
+            return path;
```

Comment:
Why return `path` here instead of something like `nil` with an inout error parameter filled? I also looked and nothing in the SDK actually uses this return value, only one test case. Would make more sense for this method to accept a path at white to store the envelope, then that test case could retrieve it that way; and this method could instead return a BOOL for success/fail if it would actually help anywhere up the callstack, or just void.

I'm also guessing here but it looks like we only need the `@synchronized` on line 315 because of that call to `uniqueAscendingJsonName`? If so, moving that out reduces the scope of this critical sector and should increase throughput.

Reply:
> I'm also guessing here but it looks like we only need the @synchronized on line 315 because of that call to uniqueAscendingJsonName? If so, moving that out reduces the scope of this critical sector and should increase throughput.

Sadly, that's not possible, because we `handleEnvelopesLimit` at the end of the method needs to be also in the synchronized block. If the max number of elements is reached, the method iterates over all envelopes and migrates the sessions init. There is a bunch of IO happening there.

Reply:
I made method void, cause as you pointed out we only use the return value for testing and we can get the latest stored envelope in tests via `sut.getAllEnvelopes().last?.path`. Thanks for pointing that out.